features/kafka/producer/src/main/java/org/opennms/features/kafka/producer/collection/KafkaPersisterFactory.java

```diff
@@ -0,0 +1,114 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.kafka.producer.collection;
+
+import java.io.IOException;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.Properties;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.common.serialization.ByteArraySerializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.opennms.features.kafka.producer.OpennmsKafkaProducer;
+import org.opennms.netmgt.collection.api.Persister;
+import org.opennms.netmgt.collection.api.PersisterFactory;
+import org.opennms.netmgt.collection.api.ServiceParameters;
+import org.opennms.netmgt.rrd.RrdRepository;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KafkaPersisterFactory implements PersisterFactory {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaPersisterFactory.class);
+    private CollectionSetMapper collectionSetMapper;
+
+    private KafkaProducer<String, byte[]> producer;
+    private ConfigurationAdmin configAdmin;
+
+    @Override
+    public Persister createPersister(ServiceParameters params, RrdRepository repository, boolean dontPersistCounters,
+            boolean forceStoreByGroup, boolean dontReorderAttributes) {
+        return createPersister(params, repository);
+    }
+
+    @Override
+    public Persister createPersister(ServiceParameters params, RrdRepository repository) {
+        KafkaPersister persister = new KafkaPersister();
+        persister.setCollectionSetMapper(collectionSetMapper);
+        persister.setProducer(producer);
+        return persister;
+    }
+
+    public void init() throws IOException {
+        // Create the Kafka producer
+        final Properties producerConfig = new Properties();
+        final Dictionary<String, Object> properties = configAdmin
+                .getConfiguration(OpennmsKafkaProducer.KAFKA_CLIENT_PID).getProperties();
+        if (properties != null) {
+            final Enumeration<String> keys = properties.keys();
+            while (keys.hasMoreElements()) {
+                final String key = keys.nextElement();
+                producerConfig.put(key, properties.get(key));
+            }
+        }
+        // Overwrite the serializers
+        producerConfig.put("key.serializer", StringSerializer.class.getCanonicalName());
```

Comment:
Can we use the existing constants for "key.serializer" and "value.serializer" here?